src/Model.php

```diff
@@ -0,0 +1,589 @@
+		// that is already in this database using the current IDs in this "where"
+		// clause to only update this model. Otherwise, we'll just insert them.
+		if ($this->exists) {
+			$saved = $this->isDirty() ?
+				$this->performUpdate($query) : true;
+		}
+
+		// If the model is brand new, we'll insert it into our database and set the
+		// ID attribute on the model to the value of the newly inserted row's ID
+		// which is typically an auto-increment value managed by the database.
+		else {
+			$saved = $this->performInsert($query);
+		}
+
+		// If the model is successfully saved, we need to do a few more things once
+		// that is done. We will call the "saved" method here to run any actions
+		// we need to happen after a model gets successfully saved right here.
+		if ($saved) {
+			$this->syncOriginal();
+		}
+
+		return $saved;
+	}
+
+	/**
+	 * Perform a model insert operation.
+	 *
+	 * @param  Query  $query
+	 * @return bool
+	 */
+	 protected function performInsert(Query $query)
+	 {
+		 if (empty($this->attributes)) {
+			 return true;
+		 }
+
+		 $success = $query->insert();
+
+		 if ($success){
+			 $this->exists = true;
+		 }
+
+		 return $success;
+	 }
+
+	/**
+	 * Perform a model insert operation.
+	 *
+	 * @param  Query  $query
+	 * @return bool
+	 */
+	 protected function performUpdate($query)
+	 {
+		 if (empty($this->attributes)) {
+			 return true;
+		 }
+
+		 $success = $query->update();
+
+		 return $success;
+	 }
+
+	/**
+	 * Perform a model insert operation.
+	 *
+	 * @param  Query  $query
+	 * @return bool
+	 */
+	protected function performDelete($query)
+	{
+		$success = $query->delete($this);
+
+		if ($success)
+		{
+			$this->exists = false;
+		}
+
+		return $success;
+	}
+
+
+	/**
+	 * Get a new query builder for the model's table.
+	 *
+	 * @return Query
+	 */
+	public function newQuery()
+	{
+		return new Query($this->db->getQuery(true), $this->db, $this);
+	}
+
+	/**
+	 * Handle dynamic method calls into the model.
+	 *
+	 * @param  string  $method
+	 * @param  array  $parameters
+	 * @return mixed
+	 */
+	public function __call($method, $parameters)
+	{
+		if (in_array($method, ['increment', 'decrement'])) {
+			return $this->$method(...$parameters);
+		}
+
+		return $this->newQuery()->$method(...$parameters);
+	}
+
+	/**
+	 * Handle dynamic static method calls into the method.
+	 *
+	 * @param  string  $method
+	 * @param  array  $parameters
+	 * @return mixed
+	 */
+	public static function __callStatic($method, $parameters)
+	{
+		return (new static)->$method(...$parameters);
+	}
+
+	/**
+	 * Create a new model instance that is existing.
+	 *
+	 * @param  array  $attributes
+	 * @param  string|null  $connection
+	 * @return static
+	 */
+	public function newFromBuilder($attributes = [], $connection = null)
+	{
+		$model = $this->newInstance([], true);
+
+		$model->setRawAttributes((array) $attributes, true);
+
+		$model->setDriver($connection ?: $this->getDriver());
```

Comment:
I think you want to pass the database object around here instead of the driver type